### DISCUSSION TITLE
Loading local image files

### DISCUSSION THREAD

**Discussion by AlexandreRangel** [cables-gl/cables_docs#859] [Answers & Questions]
Using cables standalone, is this the right way to load local images?
I need to compose the path and then dynamically load the chosen images from a different paths in the file system.

![image](https://github.com/user-attachments/assets/a724dbf5-4697-4b11-b36c-182948301680)

**Comment by steam0r** [comment]
try composing a file url, something like: `file://J:/Dropbox/artist-forest-anims/myimage.png`

you can get started by dragging a texture to your patch then copying the url and start composing from this

**Comment by AlexandreRangel** [comment]
Thank you, @steam0r ! The **file://** prefix fixed it!

**Comment by AlexandreRangel** [comment]
But then I saved the project file and it stopped working !?

**Reply by steam0r** [reply]
what do you mean, stopped working? this seems to be (weirdly) okay for me:

![368167791-eb1ee7a5-56a0-4a90-b946-dded9c2c268f](https://github.com/user-attachments/assets/fc774e86-bde1-4f79-9801-33294963dab1)

**Comment by AlexandreRangel** [comment]
It started giving the same error message as before.

![image](https://github.com/user-attachments/assets/045c60df-bdd0-4955-b147-44d2457cb508)

**Reply by steam0r** [reply]
not sure where you are getting the windows/system32 from, but make sure your file-url starts with file:// not file:\\

**Comment by AlexandreRangel** [comment]
Weird, right? The path should be as the node on the right ( file:\\ )

**Reply by steam0r** [reply]
should be file:// (forward slashes), otherwise i think someone down the line thinks this is a local file or something

**Comment by AlexandreRangel** [comment]
Tried forward and backward slashes. Same result.
Smells like some bug, because it worked for a couple of hours and I can't make it work again.

**Reply by steam0r** [reply]
can you share your .cables file, i will then try to take a look...as reduced as possible, please

**Comment by AlexandreRangel** [comment]
I deleted the file:// bit from the path, then it worked. Then typed the file:// again on the string.
Now it's working.

Then I quit cables and when opening the file, it's broken again.
The trick of editing the path fixes it again (clearing and re adding the file:// part).

Do you still want the .cables file?

**Reply by steam0r** [reply]
heh, yes, please

**Reply by AlexandreRangel** [reply]
Here's the simplified file:
[Slideshow_A_03_simple.zip](https://github.com/user-attachments/files/17031040/Slideshow_A_03_simple.zip)

![image](https://github.com/user-attachments/assets/b3eaebe6-e202-4417-8da6-bdcefacaea72)

**Reply by steam0r** [reply]
- your "index" is set to 0.15, this has to be integer
- readdir should get a directory...not a file url, try the outputs of the SystemDirs op
- texture then should get the file:// url

here is a quick example of a slideshow through all your images in your systems download folder:

[simple_slideshow.cables.zip](https://github.com/user-attachments/files/17044369/simple_slideshow.cables.zip)

**Comment by AlexandreRangel** [comment]
Great! Thank you very much, @steam0r !!
